### PR TITLE
[ADD][14.0] partner_mobile_unique

### PR DESCRIPTION
**Pull request by AshishHirapara**
initial commit.
This module validates the field ``mobile`` in the model
``res.partner`` for duplication.

As part of the validation, this module throws an error on the partner form view if another partner has the same mobile (Optionally).

### CODE REVIEW

**Review by francesco-ooops — APPROVED** [review]
functional ok!

---

For email validation there are currently two modules in this repo: 

[partner_email_duplicate_warn](https://github.com/OCA/partner-contact/tree/14.0/partner_email_duplicate_warn) warning of contact with duplicate email

[partner_email_check](https://github.com/OCA/partner-contact/tree/14.0/partner_email_check) avoiding creation of contact with duplicate email

for mobile validation there is only: 

[partner_mobile_duplicate_warn](https://github.com/OCA/partner-contact/tree/14.0/partner_mobile_duplicate_warn) warning of contact with duplicate mobile

this module avoids creation of contact with duplicate mobile.

**Review by SicurSam — APPROVED** [review]
functional ok!

**Comment by francesco-ooops** [comment]
@OCA/crm-sales-marketing-maintainers can you take a look? thanks!

**Review by rafaelbn — COMMENTED** [review]
Hello! Thank you for this PR!

I would recomend another name for this module as "check" don't specify what it checks, it's too ambiguous.

Followinf https://github.com/OCA/partner-contact/tree/14.0/partner_vat_unique I would recommend

`partner_mobile_unique`

Thanks
Regards

File: partner_mobile_check/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Copyright 2020 Ashish (Ooops) <https://ooops404.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+{
+    "name": "Partner Mobile Duplicate Checker",
```
Comment (francesco-ooops) [inline partner_mobile_check/__manifest__.py]: "name": "Partner Mobile Format and Duplicate Checker",

**Comment by francesco-ooops** [comment]
Hi @rafaelbn , as the module is basically a clone of [partner_email_check](https://github.com/OCA/partner-contact/tree/14.0/partner_email_check) for field "mobile", we thought of using the same technical name and expand description in module name

ps: can you also take a look at https://github.com/OCA/partner-contact/pull/1352 ? thanks!

**Comment by rafaelbn** [comment]
Hello @francesco-ooops ,

partner_email_check check the field, if it is a correct email no if the email is unique preventing to create another partner with the same email

`partner_email_check`
> This module validates and normalizes the field email in the model res.partner.

It not similar to this one.

The similar is;

`partner_mobile_unique`
> Module to make the VAT number unique for customers and suppliers. Will not consider empty VATs as duplicated.

`partner_mobile_check`
> Restriction on partner creation if another partner has the same mobile

Maybe I missing something, I'm reading and make unique something is not equal to check a field.

😄 Thank you!

**Comment by francesco-ooops** [comment]
@rafaelbn you're correct, 

partner_email_check normalizes, validates and avoids duplicate emails

this module only normalizes and avoids duplicate mobile numbers

we will rename it partner_mobile_unique

File: partner_mobile_check/__manifest__.py (hunk 1 above)
Comment (AshishHirapara) [reply partner_mobile_check/__manifest__.py]: done

**Comment by francesco-ooops** [comment]
@rafaelbn is it good now? :)

**Comment by francesco-ooops** [comment]
@OCA/crm-sales-marketing-maintainers good for merge?

**Comment by francesco-ooops** [comment]
@rafaelbn gentle reminder :)

**Comment by rafaelbn** [comment]
Hello test are failing maybe...

/ocabot rebase

**Review by rafaelbn — APPROVED** [review]
/ocabot merge nobump

imagen

**Comment by rafaelbn** [comment]
@AshishHirapara @francesco-ooops 

```
> 2022-10-18 12:20:10,922 251 ERROR odoo odoo.addons.partner_data_vies_populator.tests.test_partner_data_vies_populator: FAIL: TestPartnerCreateByVAT.test_company_vat_change
> Traceback (most recent call last):
>   File "/__w/partner-contact/partner-contact/partner_data_vies_populator/tests/test_partner_data_vies_populator.py", line 40, in test_company_vat_change
>     self.assertEqual(partner_form.street, self.sample_1["address"])
> AssertionError: False != 'Chaussée De Namur 40 1367 Ramillies'
>  
> 2022-10-18 12:20:10,924 251 INFO odoo odoo.addons.partner_data_vies_populator.tests.test_partner_data_vies_populator: Starting TestPartnerCreateByVAT.test_create_from_vat1 ... 
> 2022-10-18 12:20:12,767 251 WARNING odoo odoo.addons.partner_data_vies_populator.models.res_partner: Failed to query VIES: MS_MAX_CONCURRENT_REQ 
> 2022-10-18 12:20:12,775 251 INFO odoo odoo.addons.partner_data_vies_populator.tests.test_partner_data_vies_populator: ====================================================================== 
> 2022-10-18 12:20:12,775 251 ERROR odoo odoo.addons.partner_data_vies_populator.tests.test_partner_data_vies_populator: FAIL: TestPartnerCreateByVAT.test_create_from_vat1
> Traceback (most recent call last):
>   File "/__w/partner-contact/partner-contact/partner_data_vies_populator/tests/test_partner_data_vies_populator.py", line 26, in test_create_from_vat1
>     self.assertEqual(partner_form.name, self.sample_1["name"])
> AssertionError: '' != 'SA ODOO'
> + SA ODOO
```

**Comment by pedrobaeza** [comment]
This is a transient error from the VIES data populator. Just retry.

**Comment by francesco-ooops** [comment]
@rafaelbn can you retry? thanks!

**Pull request merged**